class.jetpack-gutenberg.php

```diff
@@ -89,6 +102,52 @@ class Jetpack_Gutenberg {
 	 */
 	private static $availability = array();
 
+	/**
+	 * Check to see if a block is able to run with current wp or gutenburg plugin versions
+	 *
+	 * @param string $core_wp_version The minimum wp core version needed to run the block.
+	 * @param string $plugin_version The minimum gutenberg plugin version needed to run the block.
+	 *
+	 * @since 8.4.0
+	 *
+	 * @return boolean True if the block is gated due to current wp or plugin versions.
+	 */
+	public static function is_block_version_gated( $core_wp_version, $plugin_version ) {
+		global $wp_version;
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			return false;
+		}
+
+		$current_plugin_version = self::get_plugin_version();
+
+		if ( ! empty( $current_plugin_version ) ) {
+			return $current_plugin_version < $plugin_version;
+		}
+
+		return version_compare( $wp_version, $core_wp_version, '<' );
+	}
+
+	/**
+	 * Get the version of gutenberg plugin if it is installed
+	 *
+	 * @since 8.4.0
+	 *
+	 * @return string Verion number of installed gutenberg plugin.
+	 */
+	public static function get_plugin_version() {
+		if ( is_file( WP_PLUGIN_DIR . '/gutenberg/gutenberg.php' ) ) {
+			if ( ! function_exists( 'get_plugin_data' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/plugin.php';
+			}
+			$plugin_data = get_plugin_data( WP_PLUGIN_DIR . '/gutenberg/gutenberg.php' );
+
+			if ( isset( $plugin_data['Version'] ) ) {
+				return $plugin_data['Version'];
+			}
+		}
+	}
```

Comment:
I assume other plugins may want to fetch Gutenberg's version as well; I wonder if it may be worth suggesting a change to the Gutenberg plugin directly, see if they'd be interested in adding a constant just like we do in Jetpack:
https://github.com/Automattic/jetpack/blob/5b94682c7c85f56e57278490bb38aa5213d141d6/jetpack.php#L18

It may make things easier and faster for everyone.

Reply:
I can't find the PR where we were discussing this, but there is an equivalent `GUTENBERG_VERSION` constant, bet that's only available in production builds. There's a boolean `GUTENBERG_DEVELOPMENT_MODE` constant otherwise, so we were investigating whether we could check for that being set, or the correct version.

Reply:
Found it! https://github.com/Automattic/jetpack/pull/13999#discussion_r377874798

Reply:
Thanks @pablinos  - my Monday brain had forgotten all about this conversation. Using those constants simplifies things quite a bit.